lib/Service/WhiteboardLibraryService.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Whiteboard\Service;
+
+use JsonException;
+use OCP\Files\File;
+use OCP\Files\GenericFileException;
+use OCP\Files\IRootFolder;
+use OCP\Files\NotFoundException;
+use OCP\Files\NotPermittedException;
+use OCP\Lock\LockedException;
+
+final class WhiteboardLibraryService {
+	private IRootFolder $rootFolder;
+
+	public function __construct(IRootFolder $rootFolder) {
+		$this->rootFolder = $rootFolder;
+	}
+
+	/**
+	 * @throws NotPermittedException
+	 * @throws GenericFileException
+	 * @throws LockedException
+	 * @throws JsonException
+	 */
+	public function getUserLib(string $uid): array {
+		// Get the .excalidrawlib files from the /Templates directory
+		$userFolder = $this->rootFolder->getUserFolder($uid);
+		$templatesFolder = $userFolder->get('Templates');
```

Comment:
We can switch this to https://github.com/nextcloud/server/blob/5c7216f3fde391699e910d367efe7405c7d930c1/lib/public/Files/Template/ITemplateManager.php#L39 so that it also would use template directories configured by the user.